src/Orleans/IDs/UniqueKey.cs

```diff
@@ -114,7 +114,7 @@ private static UniqueKey NewKey(ulong n0, ulong n1, Category category, long type
             // in the string representation of a key, we grab the least significant half of n1.
             // therefore, if n0 is non-zero and n1 is 0, then the string representation will always be
             // 0x0 and not useful for identification of the grain.
-            if (n1 == 0 && n1 != 0)
+            if (n1 == 0 && n0 != 0)
                 throw new ArgumentException("n0 cannot be zero unless n1 is non-zero.", "n0");
```

Comment:
It seems to me this whole check is wrong, and that it only survived because of the incorrect condition that is always `fasle`. It is totally legal for both `n0` and `n1` to be 0. For example, the very next overload `NewKey(long longKey, ` does
```csharp
var n1 = unchecked((ulong)longKey);
return NewKey(0, n1, category, typeData, keyExt);
```

where longKey can be zero.

Unless I'm overlooking something, I think we should remove this check altogether.